lib/puppet/type/netapp_qtree.rb

```diff
@@ -8,23 +8,54 @@
   ensurable
 
   newparam(:name) do
-    desc "The qtree name."
     isnamevar
+    desc "The qtree name."
     validate do |value|
-      unless value =~ /^[\w\-]+$/
+      unless (value =~ /^[\w\-\.]+$/) or (value =~ /^\/\w+\/[\w\-\.]+$/)
          raise ArgumentError, "%s is not a valid qtree name." % value
       end
     end
   end
 
   newparam(:volume) do
     desc "The volume to create qtree against."
+    defaultto do
+      if @resource[:name].count('/') == 2
+        @resource[:name].split('/')[1] 
+      else
+        ""
+      end
+    end
```

Comment:
@hunner  can you help me with this?
After some more tests it seems that I have those working. I had to change name and volume from param to property and it seems that now it's working correctly in all the cases.
I'll need to do some more checking but seems that setting them this way shold works:
```
  def self.title_patterns
    [
      [
        /^([\w\-\.]+)$/,
        [
          [:name],
        ],
      ],
      [   
        %r{^/(\w+)/([\w\-\.]+)$},
        [
          [:volume],
          [:name],
        ],
      ],
    ]
  end

  newproperty(:volume, :namevar => true) do
    desc "The volume to create qtree against."
     validate do |value|
      unless (value =~ /^\w+$/)
         raise ArgumentError, "%s is not a valid Volume name." % value
      end
     end
  end
  
  newproperty(:name, :namevar => true) do
    desc "The name of the qtree to create."
     validate do |value|
      unless (value =~ /^[\w\-\.]+$/)
         raise ArgumentError, "%s is not a valid qtree name." % value
      end
     end
  end

```
So it will validate the value if passed alone and use the regex to assign default values.